src/actions/route_helpers.cr

```diff
@@ -1,68 +1,67 @@
 require "./errors/*"
 
-module DppmRestApi::Actions
-  # A series of utility functions that are useful to all of the Action blocks
-  module RouteHelpers
-    macro fmt_route(route = "", namespace = false)
-    {{ "/" + @type.stringify
-         .downcase
-         .gsub(/^DppmRestApi::Actions::/, "")
-         .gsub(/::/, "/") }} {% if namespace %} + "/:namespace" {% end %} + ( {{route}} || "")
-    end
+# A series of route utilities functions that are useful to all of the Action blocks.
+module DppmRestApi::Actions::RouteHelpers
+  macro included
+  ROOT_PATH = {{ "/" + @type.stringify.downcase.split("::").last }}
 
-    # the "last" or -1st block argument is selected because context is always the
-    # argument -- either the only or the second of two
-    {% for method in %w(get post put delete ws options) %}
-    macro relative_{{method.id}}(route, &block)
-      {{method.id}} fmt_route(\{{route}}) do |\{{block.args.splat}}|
-        namespace = \{{block.args[-1]}}.params.query["namespace"]? || DPPM::Prefix.default_group
-        \{{block.body}}
-      rescue ex : Actions::Exception
-        raise ex
-      rescue ex
-        # Catch non-HTTPStatusError exceptions, and throw an InternalServerError
-        # with the original error as the cause.
-        raise InternalServerError.new \{{block.args[0]}}, cause: ex
-      end
+  # Transform a relative route to an absolute one from the `ROOT_PATH`.
+  def fmt_route(relative_route : String? = nil, namespace : Bool = false)
+    ROOT_PATH + (namespace ? "/:namespace" : "") + relative_route.to_s
+  end
+
+  {% for method in %w(get post put delete ws options) %}
+  # Perform a relative {{method.id}} from the `ROOT_PATH`.
+  def relative_{{method.id}}(route : String?, &block : HTTP::Server::Context ->)
```

Comment:
Why??

Reply:
Because better to avoid unneeded macros. A classic method do the job.

Reply:
> better to avoid unneeded macros

What makes you say that? I don't see anything wrong with making a liberal use of macros. That said, after giving it some thought, in this case it does neaten things up a bit.

Reply:
I disagree, macros have strong downsides. That why they must not be abused, and **only** used when strongly needed.
- harder to debug, specially compile time errors
- slower compile times
- lazy typing, no type for arguments, no signature
- unwanted side effects (modifying variables in the current scope)